### PR TITLE
Avoid unnecessary re-renderings in dashboard - dashboard controls

### PR DESCRIPTION
**Pull request by uladzimirdev**
### Description

After https://github.com/metabase/metabase/pull/42777 we have several unnecessary re-renderings related to the props which are re-created on DashboardControls level, this PR uses memoization to avoid having new references for the passed props

### Before
image

### After
image

### How to verify

CI is green

### CODE REVIEW

File: frontend/src/metabase/dashboard/hoc/DashboardControls.tsx [hunk 1]
```diff
@@ -12,8 +13,9 @@ import type {
   DashboardControlsProps,
 } from "./types";
 
-/* This contains some state for dashboard controls on both private and embedded dashboards.
- * It should probably be in Redux?
+/**
```
Comment (uladzimirdev) [inline frontend/src/metabase/dashboard/hoc/DashboardControls.tsx:16]: IDE requires two `**` to parse `@deprecated` correctly

File: frontend/src/metabase/dashboard/hoc/DashboardControls.tsx [hunk 2]
```diff
@@ -89,5 +91,5 @@ export const DashboardControls = <T extends DashboardControlsProps>(
     );
   }
 
-  return DashboardControlsInner;
+  return memo(DashboardControlsInner);
```
Comment (uladzimirdev) [inline frontend/src/metabase/dashboard/hoc/DashboardControls.tsx:94]: if parent is changed, we don't need to re-render

**Review by oisincoveney — APPROVED** [review]
Love these changes. Seeing modifications like this is really helping me learn a lot more about what I can do better. Nice!

File: frontend/src/metabase/dashboard/hooks/use-dashboard-refresh-period.ts [hunk 3]
```diff
@@ -16,33 +17,41 @@ export const useDashboardRefreshPeriod = ({
   const elapsedHook = useRef<((elapsed: number | null) => void) | null>(null);
   const elapsed = useRef<number | null>(0);
 
-  const setRefreshElapsedHook = (hook: (elapsed: number | null) => void) => {
-    elapsedHook.current = hook;
-  };
+  const setRefreshElapsedHook = useCallback(
+    (hook: (elapsed: number | null) => void) => {
+      elapsedHook.current = hook;
+    },
```
Comment (oisincoveney) [inline frontend/src/metabase/dashboard/hooks/use-dashboard-refresh-period.ts:23]: Do we need the useCallback here? Curious since there's no dependencies

Comment (uladzimirdev) [reply frontend/src/metabase/dashboard/hooks/use-dashboard-refresh-period.ts:23]: every time you run this hook, new function will be created and new reference will be returned, so yes

Comment (uladzimirdev) [reply frontend/src/metabase/dashboard/hooks/use-dashboard-refresh-period.ts:23]: see the screenshot at `Before` section in the description, it shows that this function had a new reference

Comment (oisincoveney) [reply frontend/src/metabase/dashboard/hooks/use-dashboard-refresh-period.ts:23]: Ah yes of course. Thanks 😄

**Comment by uladzimirdev** [comment]
`no-backport` as original PR which introduced those hooks wasn't backported

**Pull request merged**